### ISSUE TITLE
Can't reach the first key in outline

### ISSUE THREAD

**Issue by dangh** [zed-industries/zed#4621]
### Check for existing issues

- [X] Completed

### Describe the feature

I can't reach the first "type" key in this JSON document with outline:

```json
{
	"type": "string",
	"at": "some where",
	"nested": {
		"type": "number"
	}
}
```

- Type `t`, the cursor in the buffer jump to "at".
- Type `ype`, zed focus the second "type".

I thought Zed does an incremental search so it jump to second "type" after focusing "at", but even typing `ype` from the beginning will jump to second "type".

**Comment by notpeter** [comment]
I can no longer reproduce this and I believe it has been fixed.
If this is incorrect, please feel free to comment and I'm happy to reopen.
Thanks for reporting!

**Comment by dangh** [comment]
Hi @notpeter the issue is still there. The problem is Zed always favors the second `type` so I have to use the arrow key to reach the first `type`. I would prefer the opposite.

**Comment by notpeter** [comment]
I still can't seem to reproduce. Can you provide a short screen recording of what you're doing and/or provide silly-detailed ELI5 set of instructions.  I'm sure you're seeing this, but the different ways I try to something similar aren't triggering the behavior you describe (for me).

**Comment by notpeter** [comment]
@dangh Bump. Any chance you demonstrate what you're seeing so we can try and reproduce/fix?

**Comment by dangh** [comment]
Hi @notpeter sorry for the late response. The issue title is misleading. The actual issue is when I type the key in outline panel, Zed favor the last matched entry instead of the first one. You can see the issue in the record below.

- When I type `t` it select "at" instead of "type".
- When I type `type` it select the nested "type" instead of the first entry.

https://github.com/user-attachments/assets/52651f17-f9d3-4ae2-a6a3-76fb58bab1f9

In VS Code and Sublime Text, it select the first entry and I have to use the arrow keys to select the others.

https://github.com/user-attachments/assets/7d2ade57-b2eb-40e2-9334-4fce17459917

https://github.com/user-attachments/assets/32cc4b3c-a1bc-42e9-98b6-907c3b260521

I would prefer the behaviour of VS Code and Sublime Text here.

**Comment by beniaminzagan** [comment]
@notpeter the issue is that he types ty and the nested type is chosen instead of the first occurance, i was able to reproduce in zed preview 175.1

**Comment by dangh** [comment]
it still happened on Zed Preview 0.213.3 
7db9d5ff09aa11181e077756dc42f15123739ba8

**Comment by miguelraz** [comment]
Sorry all for the noise, the bot got an issue upstream and went haywire: https://github.com/actions/stale/issues/1302.

**Comment by dangh** [comment]
It's better in Zed Preview 1.3
Typing `type` will select the outer `type`. But typing `t` it still select `at`.

**Comment by dangh** [comment]
Happy to report that it is fixed.

Zed Preview 1.10.0 
Commit: 8d56867088a4d4269bed518cdb1f913a037179de
Version: 1.10.0+preview.315.8d56867088a4d4269bed518cdb1f913a037179de
